src/App/Tracker/Controller/Hooks/Listeners/JoomlacmsPullsListener.php

```diff
@@ -334,169 +262,73 @@ protected function checkMasterBranch($hookData, Github $github, Logger $logger,
 	/**
 	 * Checks for a PR-<branch> label
 	 *
-	 * @param   object  $hookData  Hook data payload
-	 * @param   Github  $github    Github object
-	 * @param   Logger  $logger    Logger object
-	 * @param   object  $project   Object containing project data
+	 * @param   object       $hookData  Hook data payload
+	 * @param   Github       $github    Github object
+	 * @param   Logger       $logger    Logger object
+	 * @param   object       $project   Object containing project data
+	 * @param   IssuesTable  $table     Table object
 	 *
 	 * @return  void
 	 *
 	 * @since   1.0
 	 */
-	protected function checkPullLabel($hookData, Github $github, Logger $logger, $project)
+	protected function checkPullLabels($hookData, Github $github, Logger $logger, $project, IssuesTable $table)
 	{
 		// Set some data
-		$issueLabel       = 'PR-' . $hookData->pull_request->base->ref;
+		$prLabel          = 'PR-' . $hookData->pull_request->base->ref;
 		$languageLabel    = 'Language Change';
 		$addLabels        = array();
-		$prLabelSet       = false;
-		$languageLabelSet = false;
+		$removeLabels     = array();
+		$prLabelSet       = $this->checkLabel($hookData, Github $github, Logger $logger, $project, IssuesTable $table, $prLabel);
+		$languageChange   = $this->checkLanguageChange($files);
+		$languageLabelSet = $this->checkLabel($hookData, Github $github, Logger $logger, $project, IssuesTable $table, $languageLabel);
 
-		// Get the labels for the pull's issue
-		try
+		// Add the issueLabel if it isn't already set
+		if (!$prLabelSet)
 		{
-			$labels = $github->issues->get($project->gh_user, $project->gh_project, $hookData->pull_request->number)->labels;
+			$addLabels[] = $prLabel;
 		}
-		catch (\DomainException $e)
-		{
-			$logger->error(
-				sprintf(
-					'Error retrieving labels for GitHub item %s/%s #%d - %s',
-					$project->gh_user,
-					$project->gh_project,
-					$hookData->pull_request->number,
-					$e->getMessage()
-				)
-			);
 
-			return;
-		}
-
-		// Check for labels if there are already labels attached to the item
-		if (count($labels) > 0)
+		if ($languageChange && !$languageLabelSet)
 		{
-			foreach ($labels as $label)
-			{
-				if (!$prLabelSet && $label->name == $issueLabel)
-				{
-					$logger->info(
-						sprintf(
-							'GitHub item %s/%s #%d already has the %s label.',
-							$project->gh_user,
-							$project->gh_project,
-							$hookData->pull_request->number,
-							$issueLabel
-						)
-					);
-
-					$prLabelSet = true;
-				}
-
-				if (!$languageLabelSet && $label->name == $languageLabel)
-				{
-					$logger->info(
-						sprintf(
-							'GitHub item %s/%s #%d already has the %s label.',
-							$project->gh_user,
-							$project->gh_project,
-							$hookData->pull_request->number,
-							$languageLabel
-						)
-					);
-
-					$languageLabelSet = true;
-				}
-			}
+			$addLabels[] = $languageLabel;
+			$this->addLabels($hookData, Github $github, Logger $logger, $project, IssuesTable $table, $addLabels);
```

Comment:
You should remove type hinting here. And also do not forget to remove $table as it is not used.